src/compiler/parser.ts

```diff
@@ -4887,15 +4891,22 @@ namespace ts {
             return decorators;
         }
 
-        function parseModifiers(): ModifiersArray {
+        function parseModifiers(permitConstAsModifier: boolean = false): ModifiersArray {
             let flags = 0;
             let modifiers: ModifiersArray;
             while (true) {
                 const modifierStart = scanner.getStartPos();
                 const modifierKind = token;
 
-                if (!parseAnyContextualModifier()) {
-                    break;
+                if (permitConstAsModifier && token === SyntaxKind.ConstKeyword) {
+                    if (!tryParse(nextTokenIsOnSameLineAndCanFollowModifier)) {

```

Comment:
Add a comment above:

```ts
// We need to ensure that any subsequent modifiers appear on the same line
// so that when 'const' is a standalone declaration, we don't issue an error.
```